tools/generate_property_translations.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+#
+# Generates translation strings from gecko property files for lupdate
+# to process
+
+import os, os.path, sys, re, dataclasses, functools
+
+PREFIX = "sailfish_components_webview_popups-la"
+
+class Generator:
+    RE = re.compile(r"(?P<name>[\w-]+)\s*=\s*(?P<text>.*)\s*")
+    SPLITTER = re.compile(r"([A-Za-z0-9][a-z0-9]*)")
+    PLACEHOLDER = re.compile(r"%(?:([0-9]+)\$)?S")
+
+    def __init__(self, file):
+        self.file = file
+
+    @property
+    @functools.cache
+    def name(self):
+        """Basename from self.file without file extension and
+        with dashes replaced by underscores"""
+        return os.path.splitext(os.path.basename(self.file))[0].replace("-", "_")
+
+    @staticmethod
+    def _convert_key(name):
+        """Convert from camelCase to snake_case"""
+        return "_".join(Generator.SPLITTER.findall(name)).lower()
+
+    @staticmethod
+    def _convert_text(text):
+        """Replace gecko's placeholders with Qt's and
+        remove access key hints"""
+        def replacer(match):
+            return "%1" if match.group(1) is None else "%" + match.group(1)
+        new = Generator.PLACEHOLDER.sub(replacer, text)
+        while new != text:
+            text = new
+            new = Generator.PLACEHOLDER.sub(replacer, text)
+        return new.replace("&", "")
+
+    def process_line(self, line):
+        match = self.RE.match(line)
+        if match is not None:
+            return Translation(self.name,
+                               self._convert_key(match.group('name')),
+                               self._convert_text(match.group('text')))
+
+    def __iter__(self):
+        try:
+            with open(self.file) as file:
+                for line in file:
+                    result = self.process_line(line)
+                    if result is not None:
+                        yield result
+        except UnicodeDecodeError:
+            print(f"Invalid character in file: {self.file}", file=sys.stderr)
+            raise
+
+@dataclasses.dataclass
+class Translation:
+    name: str
+    key: str
+    text: str
+
+    def __str__(self):
+        return f"""
```

Comment:
Hmm... wondering if we should actually have the original key here...

//: The original translation key - {self.original_key}

Reply:
Added this too.